### PR TITLE
Fix nil pointer bug if `bot_protection.requirements.signupOrLogin` is empty

### PR DESCRIPTION
**Pull request by pkong-ds**
ref sentry https://oursky.slack.com/archives/C01P0NCS3F1/p1723522268603209

### CODE REVIEW

**Comment by pkong-ds** [comment]
as per https://oursky.slack.com/archives/C01P0NCS3F1/p1723526303743499?thread_ts=1723522268.603209&cid=C01P0NCS3F1


> Can you audit your recent additions and see if there are similar patterns?

I added an extra bool return for all fns in `pkg/lib/authenticationflow/declarative/generate_config_bot_protection.go`

I also found `GetBotProtectionData` in `pkg/lib/authenticationflow/declarative/data_bot_protection.go` returning nil; But I decided not to change this, due to
   1. convenience to inline the function call in struct construction
   2. massive refactoring required
   3. Checked most usage expected nil retur

Please let me know if I should change that too 

```
func GetBotProtectionData(authflowCfg *config.AuthenticationFlowBotProtection, appCfg *config.BotProtectionConfig) *BotProtectionData {
```

File: pkg/lib/authenticationflow/declarative/generate_config_bot_protection.go [hunk 1]
```diff
@@ -5,59 +5,59 @@ import "github.com/authgear/authgear-server/pkg/lib/config"
 func isBotProtectionEnabled(cfg *config.AppConfig) bool {
 	return cfg.BotProtection != nil && cfg.BotProtection.Enabled && cfg.BotProtection.Provider != nil
 }
-func getBotProtectionRequirementsSignupOrLogin(cfg *config.AppConfig) (botProtection *config.AuthenticationFlowBotProtection) {
+func getBotProtectionRequirementsSignupOrLogin(cfg *config.AppConfig) (botProtection *config.AuthenticationFlowBotProtection, exist bool) {
```
Comment (louischan-oursky) [inline pkg/lib/authenticationflow/declarative/generate_config_bot_protection.go:8]: We commonly call `ok bool`

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

**Comment by louischan-oursky** [comment]
> Checked most usage expected nil return

Does that mean some usage did not expected nil return?

**Comment by pkong-ds** [comment]
> Does that mean some usage did not expected nil return?

Sor, _all_ usage expected nil return

```
$ rg 'GetBotProtectionData'
pkg/lib/authenticationflow/declarative/data_create_authenticator_option.go
112:					BotProtection:  GetBotProtectionData(b.BotProtection, deps.Config.BotProtection),
139:					BotProtection:  GetBotProtectionData(b.BotProtection, deps.Config.BotProtection),
162:					BotProtection:  GetBotProtectionData(b.BotProtection, deps.Config.BotProtection),
170:					BotProtection:  GetBotProtectionData(b.BotProtection, deps.Config.BotProtection),

pkg/lib/authenticationflow/declarative/utils_bot_protection.go
15:	data := GetBotProtectionData(authflowCfg, appCfg)

pkg/lib/authenticationflow/declarative/data_authenticate_option.go
78:		BotProtection:  GetBotProtectionData(authflowBotProtectionCfg, appBotProtectionConfig),
85:		BotProtection:  GetBotProtectionData(authflowBotProtectionCfg, appBotProtectionConfig),
93:		BotProtection:  GetBotProtectionData(authflowBotProtectionCfg, appBotProtectionConfig),
100:		BotProtection:  GetBotProtectionData(authflowBotProtectionCfg, appBotProtectionConfig),
119:			BotProtection:     GetBotProtectionData(authflowBotProtectionCfg, appBotProtectionConfig),
133:			BotProtection:     GetBotProtectionData(authflowBotProtectionCfg, appBotProtectionConfig),
154:				BotProtection:     GetBotProtectionData(authflowBotProtectionCfg, appBotProtectionConfig),
166:				BotProtection:     GetBotProtectionData(authflowBotProtectionCfg, appBotProtectionConfig),

pkg/lib/authenticationflow/declarative/data_bot_protection.go
30:func GetBotProtectionData(authflowCfg *config.AuthenticationFlowBotProtection, appCfg *config.BotProtectionConfig) *BotProtectionData {

pkg/lib/authenticationflow/declarative/intent_account_recovery_flow_step_identify.go
73:			c := AccountRecoveryIdentificationOption{Identification: b.Identification, BotProtection: GetBotProtectionData(b.GetBotProtectionConfig(), deps.Config.BotProtection)}

pkg/lib/authenticationflow/declarative/data_identification.go
43:		BotProtection:  GetBotProtectionData(authflowCfg, appCfg),
50:		BotProtection:  GetBotProtectionData(authflowCfg, appCfg),
60:				BotProtection:  GetBotProtectionData(authflowCfg, appCfg),
73:		BotProtection:  GetBotProtectionData(authflowCfg, appCfg),
```

**Comment by pkong-ds** [comment]
@louischan-oursky updated, thanks!

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**